nmdc_runtime/site/translation/neon_translator.py

```diff
@@ -0,0 +1,819 @@
+                status=self._get_value_or_none(extraction_row, "qaqcStatus")
+            ),
+            processing_institution=processing_institution,
+        )
+
+    def _translate_library_preparation(
+        self,
+        library_preparation_id,
+        library_preparation_input,
+        processed_sample_id,
+        library_preparation_row,
+    ):
+        """
+        Translate a library_preparation_row DataFrame into a LibraryPreparation object.
+        """
+        processing_institution = None
+        laboratory_name = self._get_value_or_none(
+            library_preparation_row, "laboratoryName"
+        )
+        if laboratory_name is not None:
+            if re.search("Battelle", laboratory_name, re.IGNORECASE):
+                processing_institution = "Battelle"
+            elif re.search("Argonne", laboratory_name, re.IGNORECASE):
+                processing_institution = "ANL"
+
+        return nmdc.LibraryPreparation(
+            id=library_preparation_id,
+            has_input=library_preparation_input,
+            has_output=processed_sample_id,
+            start_date=self._get_value_or_none(library_preparation_row, "collectDate"),
+            end_date=self._get_value_or_none(library_preparation_row, "processedDate"),
+            processing_institution=processing_institution,
+        )
+
+    def _translate_omics_processing(
+        self,
+        omics_processing_id: str,
+        processed_sample_id: str,
+        raw_data_file_data: str,
+        omics_processing_row: pd.DataFrame,
+    ) -> nmdc.OmicsProcessing:
+        processing_institution = None
+        sequencing_facility = self._get_value_or_none(
+            omics_processing_row, "sequencingFacilityID"
+        )
+        if sequencing_facility is not None:
+            if re.search("Battelle", sequencing_facility, re.IGNORECASE):
+                processing_institution = "Battelle"
+            elif re.search("Argonne", sequencing_facility, re.IGNORECASE):
+                processing_institution = "ANL"
+
+        return nmdc.OmicsProcessing(
+            id=omics_processing_id,
+            has_input=processed_sample_id,
+            has_output=raw_data_file_data,
+            processing_institution=processing_institution,
+            ncbi_project_name=self._get_value_or_none(
+                omics_processing_row, "ncbiProjectID"
+            ),
+            omics_type=self._create_controlled_term_value(
+                omics_processing_row["investigation_type"].values[0]
+            ),
+            instrument_name=f"{self._get_value_or_none(omics_processing_row, 'sequencingMethod')} {self._get_value_or_none(omics_processing_row, 'instrument_model')}",
+            part_of="nmdc:sty-11-34xj1150",
+            name=f"Terrestrial soil microbial communities - {self._get_value_or_none(omics_processing_row, 'dnaSampleID')}",
+            type="nmdc:OmicsProcessing",
+        )
+
+    def get_database(self) -> nmdc.Database:
```

Comment:
Consider a more descriptive function name to indicate that this returns the database with new data source translated and merged in.  Also consider docstrings for complex functions like this